packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -519,9 +524,11 @@ export class CfnParser {
     if (isRef) {
       const refElement = this.options.finder.findRefTarget(refTarget);
       if (!refElement) {
-        throw new Error(`Element referenced in Fn::Sub expression with logical ID: '${refTarget}' was not found in the template`);
+        if  (!(refTarget in this.parameters)) {
```

Comment:
I don't understand this... `refTarget` cannot be in `this.parameters`, because you call `specialCaseSubRefs()` above. So this condition is always true.

In other words, this should be unchanged.